salt/orch/update.sls

```diff
@@ -0,0 +1,121 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+
+update_modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+
+# Get list of masters needing reboot
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master and G@upgrade:needs', fun='network.ip_addrs', tgt_type='compound') %}
```

Comment:
It looks like the grain we'll want will be called:

    tx_update_reboot_needed: true